### PR TITLE
Get list of weather years (if available) for a given scenario

### PR DESCRIPTION
**Pull request by johannes-maywald**
Mostly followed other patterns,
For simplicity did not add snippets to `gql.fragments.scenario`, `gql.queries.scenario_queries`, or the likes
Not sure if this is best placed in there to keep things neat?

### CODE REVIEW

**Comment by johannes-maywald** [comment]
if you want to write tests:
* should return years 2007-2016 for GB October 2023 (and likely onwards)
* empty list for other (GB) scenarios before that

File: src/origin_sdk/OriginSession.py [hunk 1]
```diff
@@ -900,3 +900,23 @@ def get_workbook_download_url(self, scenario_id: str):
...
+        """
+
+        variables = {"scenarioGlobalId": scenario_id}
+        get_weather_years = f"""
+        query ($scenarioGlobalId: String!) {{
+            getWeatherYearList(scenarioGlobalId: $scenarioGlobalId) {{
+                years
+            }}
+        }}
+        """
```
Comment (jebbin-joseph) [inline src/origin_sdk/OriginSession.py:920]: Can we please move the query to `src/origin_sdk/gql/scenario_queries`. That's where we keep all the queries.

**Review by jebbin-joseph — COMMENTED** [review]
By introducing this, we can now let users set the weatherYear for a scenario. So we also should update the type in `src/origin_sdk/type/scenario_types` to tell what weather year does.

**Review by jebbin-joseph — COMMENTED** [review]
You also need to update the SDK version in `pyproject.toml` manually. Currently I think it's `0.20.0`. So we should bump it to `0.21.0`.

**Comment by johannes-maywald** [comment]
Thanks Jebbin!
Though, not sure what you mean by this: 
> By introducing this, we can now let users set the weatherYear for a scenario. So we also should update the type in `src/origin_sdk/type/scenario_types` to tell what weather year does.

* Is this about adding it to the doc string for the `InputScenario(TypedDict)` attributes ?
* or de-commenting it from the `ScenarioType` class/dict ?

**Review by jebbin-joseph — APPROVED** [review]
LGTM. 🔥

**Pull request merged**